pkg/monitors/http/http.go

```diff
@@ -0,0 +1,179 @@
+package http
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/signalfx/golib/v3/datapoint"
+	"github.com/signalfx/signalfx-agent/pkg/core/config"
+	"github.com/signalfx/signalfx-agent/pkg/monitors"
+	"github.com/signalfx/signalfx-agent/pkg/monitors/types"
+	"github.com/signalfx/signalfx-agent/pkg/utils"
+	"github.com/sirupsen/logrus"
+)
+
+func init() {
+	monitors.Register(&monitorMetadata, func() interface{} { return &Monitor{} }, &Config{})
+}
+
+// Config for this monitor
+type Config struct {
+	config.MonitorConfig `singleInstance:"false" acceptsEndpoints:"false"`
+	Body                 string            `yaml:"body"`
+	FollowRedirects      bool              `yaml:"followRedirects" default:"true"`
+	Headers              map[string]string `yaml:"headers"`
+	Method               string            `yaml:"method" default:"GET"`
+	Timeout              int               `yaml:"timeout" default:"5"`
+	URLs                 []string          `yaml:"urls"`
+	Regex                string            `yaml:"regex"`
```

Comment:
Add comments for all of the config options (they get included into the user docs).